os/board/common/partitions.c

```diff
@@ -158,6 +158,18 @@ static int type_specific_initialize(int minor, FAR struct mtd_dev_s *mtd_part, i
 		smart_initialize(minor, mtd_part, partref);
 		partinfo->smartfs_partno = partno;
 		mtd_setpartitiontagno(mtd_part, MTD_FS);
+#ifndef CONFIG_SMARTFS_MULTI_ROOT_DIRS
+		char fs_devname[FS_PATH_MAX];
+		snprintf(fs_devname, FS_PATH_MAX, "/dev/smart%dp%d", minor, partno);
+		int ret = mksmartfs(fs_devname, false);
+		if (ret != OK) {
+			printf("ERROR: mksmartfs on %s failed errno : %d\n", fs_devname, errno);
+			/* TODO: What to do if mksmartfs fails
+			 * 	 It is not required to halt the bootup
+			 */
+		}
+#endif
+/* TODO: Support for Multiple Root Directory mounts */
```

Comment:
what if someone enables CONFIG_SMARTFS_MULTI_ROOT_DIRS in the config? does this code cause any errors? If so, I feel it's better to add some error message so that the compilation terminates if the config is selected.